extensions/azurecore/src/azureResource/tree/connectionDialogTreeProvider.ts

```diff
@@ -53,20 +53,31 @@ export class ConnectionDialogTreeProvider implements vscode.TreeDataProvider<Tre
 			return [AzureResourceMessageTreeNode.create(localize('azure.resource.tree.treeProvider.loadingLabel', "Loading ..."), undefined)];
 		}
 
-		try {
-			if (this.accounts && this.accounts.length > 0) {
-				const accountNodes: FlatAccountTreeNode[] = [];
-				for (const account of this.accounts) {
+		if (this.accounts && this.accounts.length > 0) {
+			const accountNodes: FlatAccountTreeNode[] = [];
+			const errorMessages: string[] = [];
+			for (const account of this.accounts) {
```

Comment:
Might want to put a comment here explaining that you purposely don't iterate over all accounts at the same time to avoid Azure API rate limits.